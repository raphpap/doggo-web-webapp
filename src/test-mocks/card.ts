// Types
import {Card} from 'doggo/context';

export const mockCard = (id: number): Card => ({
  attack: id,
  defense: id,
  hpLeft: id,
  hpTotal: id,
  id: `${id}`,
  image: `${id}`,
  name: `${id}`,
});
