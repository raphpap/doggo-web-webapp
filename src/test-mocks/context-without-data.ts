// Types
import {ApplicationContext, BattleStatus} from 'doggo/context';

export const mockContextWithoutData = (): ApplicationContext => ({
  actions: {
    battle: jest.fn(),
    capture: jest.fn(),
    getNextOpponent: jest.fn(),
    login: jest.fn(),
    selectBattleCard: jest.fn(),
    unselectBattleCard: jest.fn(),
  },
  state: {
    battle: {
      cardId: null,
      opponent: null,
      status: BattleStatus.NoOpponent
    },
    cards: null,
    error: null,
    loading: false
  }
});
