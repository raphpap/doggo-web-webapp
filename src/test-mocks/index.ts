export * from './card';
export * from './context-without-data';
export * from './logged-in';
