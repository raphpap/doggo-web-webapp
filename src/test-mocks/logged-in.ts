// Types
import {ApplicationContext, BattleStatus} from 'doggo/context';

// Mocks
import {mockCard, mockContextWithoutData} from '.';

export const mockLoggedIn = (): ApplicationContext => {
  const context = mockContextWithoutData();

  return {
    ...context,
    state: {
      ...context.state,
      battle: {
        ...context.state.battle,
        opponent: mockCard(99),
        status: BattleStatus.NoCardSelected
      },
      cards: [
        mockCard(1),
        mockCard(2)
      ]
    }
  };
};
