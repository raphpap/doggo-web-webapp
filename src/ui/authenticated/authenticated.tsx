// Vendor
import React, {Component} from 'react';
import {compose} from 'recompose';

// Vendor components
import {
  Redirect,
  Route,
  RouteComponentProps,
  Switch,
  withRouter
} from 'react-router-dom';

// Shared components
import Container from 'doggo/ui/@components/container';
import Content from 'doggo/ui/@components/content';
import Header from 'doggo/ui/@components/header';

// Components
import Battle from './battle';
import Capture from './capture';
import Team from './team';

// Context
import {
  withApplicationContext,
  WithApplicationContextProps
} from 'doggo/context';

// Types
interface Props extends RouteComponentProps<never> {}
type EnhancedProps = Props & WithApplicationContextProps;

const enhance = compose<EnhancedProps, Props>(
  withApplicationContext,
  withRouter
);

// Interfaces
export class Authenticated extends Component<EnhancedProps> {
  public render() {
    const {context, location} = this.props;
    const {state} = context;
    const {cards} = state;
    const isAuthenticated = !!cards;

    if (!isAuthenticated) {
      return (
        <Redirect
          to={{
            pathname: '/login',
            state: {from: location}
          }}
        />
      );
    }

    return (
      <Container>
        <Header />
        <Content>
          <Switch>
            <Route path="/capture" component={Capture} />
            <Route
              path="/team/card/:cardId"
              render={({match}) => (
                <Team cardId={match.params.cardId as string} />
              )}
            />
            <Route path="/team" component={Team} />
            <Route path="/battle" component={Battle} />
            <Route component={Capture} />
          </Switch>
        </Content>
      </Container>
    );
  }
}

export default enhance(Authenticated);
