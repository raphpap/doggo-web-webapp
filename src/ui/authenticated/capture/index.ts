export * from './capture';
export {default} from './capture';
