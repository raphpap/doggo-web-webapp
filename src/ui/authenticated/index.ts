export * from './authenticated';
export {default} from './authenticated';
