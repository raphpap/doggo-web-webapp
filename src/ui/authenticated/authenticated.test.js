// Vendor
import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

// Vendor components
import {Redirect} from 'react-router-dom';

// Components
import {Authenticated} from './authenticated';

// Mocks
import {mockContextWithoutData, mockLoggedIn} from 'doggo/test-mocks';

describe('<Authenticated />', () => {
  let wrapper;
  let context;
  let location = {state: {from: {pathname: '/current-path'}}};

  beforeEach(() => {
    context = mockContextWithoutData();
    wrapper = shallow(<Authenticated context={context} location={location} />);
  });

  it('redirects to the login when not logged in', () => {
    const redirect = wrapper.find(Redirect);
    expect(redirect.props().to.pathname).toBe('/login');
    expect(redirect.props().to.state.from).toBe(location);
  });

  it('renders correctly when logged in', () => {
    context = mockLoggedIn();
    wrapper.setProps({context});
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
