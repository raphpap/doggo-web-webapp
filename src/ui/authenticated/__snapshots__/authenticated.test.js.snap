// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Authenticated /> renders correctly when logged in 1`] = `
<Container>
  <withRouter(Header) />
  <Content>
    <Switch>
      <Route
        component={[Function]}
        path="/capture"
      />
      <Route
        path="/team/card/:cardId"
        render={[Function]}
      />
      <Route
        component={[Function]}
        path="/team"
      />
      <Route
        component={[Function]}
        path="/battle"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </Content>
</Container>
`;
